src/core/dev/ib_ctx_handler.cpp

```diff
@@ -138,12 +118,11 @@ ib_ctx_handler::ib_ctx_handler(struct ib_ctx_handler_desc *desc)
         ibv_dealloc_pd(m_p_ibv_pd);
     }
 
-#ifdef DEFINED_DPCP
     if (m_p_adapter) {
         delete m_p_adapter;
         m_p_ibv_context = NULL;
     }
-#endif /* DEFINED_DPCP */
+
     if (m_p_ibv_context) {
         ibv_close_device(m_p_ibv_context);
         m_p_ibv_context = NULL;
```

Comment:
This block can be removed, because ibv context is part of dpcp adapter and the adapter is always present.

Reply:
Looks like a bug in our existing code with dpcp.

Reply:
Maybe we should get rid of m_p_ibv_context and use (ibv_context *)adapter->get_ibv_context() in all places